pkg/kn/commands/completion_helper.go

```diff
@@ -30,6 +30,7 @@ type completionConfig struct {
 var (
 	resourceToFuncMap = map[string]func(config *completionConfig) []string{
 		"service": completeService,
+		"domain":  completeDomain,
```

Comment:
Since that's going to be the last conflict. Maybe, I'd order the list alphabetically by the resource name for the better overview.